### ISSUE TITLE
Feature request: allow to execute macro without an import

### ISSUE THREAD

**Issue by mikemix** [twigphp/Twig#3134]
How many times did you see that?

```
{% import 'macro.html.twig' as macro %}
{{ macro.something(input) }}
```

A shorthand (one liner) would be lovely, eg.:

```
{{ from 'macro.html.twig' run something(input) }}
```

or 

```
{{ macro something(input) from 'macro.html.twig' }}
```

**Comment by stof** [comment]
this would require a different syntax for the `{{ }}` syntax, as it currently takes a valid expression (any expression), while your proposal does not look like an expression (it is far from a valid expression by today's rules). And I don't see an easy way to turn that into a valid syntax (and without making the parsing ambiguous).
Thus, this would not allow removing the existing syntaxes to import macros, as being able to call macros multiple times is a valid use case that we cannot drop. So I'm not sure about that.

**Comment by mikemix** [comment]
I wasn't proposing to drop current macro import functionality. It's just a proposal, feel free to close it if it doesn't make any sense.

**Comment by stof** [comment]
Well, not dropping the current syntax means we add yet another way to use macro, which adds more complexity to Twig. That's why I mentioned that point (but I should have explained it better in my previous post).

**Comment by javiereguiluz** [comment]
Importing macros defined in the same template that uses them was very annoying (but was fixed in Twig 2.11: https://symfony.com/blog/simpler-macros-in-twig-templates). However, importing macros from external templates doesn't look that bad to me.

First, the import is explicit so you can quickly see that this is something defined elsewhere:

```twig
{% import 'macro.html.twig' as macro %}
```

Second, if you use the same macro more than once in a template (something that in my experience is common), you only have to do this:

```twig
{% import 'macro.html.twig' as macro %}
...
{{ macro.something(input) }}
{{ macro.something(other_input) }}
{{ macro.something(another_input) }}
```

However, with this proposal, you should re-import everytime:

```twig
{{ from 'macro.html.twig' run something(input) }}
{{ from 'macro.html.twig' run something(other_input) }}
{{ from 'macro.html.twig' run something(another_input) }}
```

In summary, I agree with you that for one-time use macros defined in other templates your proposal would be OK ... but that looks a too narrow use case to me.

**Comment by fabpot** [comment]
I agree with comments made by @javiereguiluz (and @stof is right as well about the syntax). Let's close. Thanks for proposing.
